idpi/src/operators/destagger.py

```diff
@@ -2,21 +2,35 @@
 
 
 def destagger(field, dim):
-    """Destagger a field."""
+    """Destagger a field.
+
+    Arguments:
+    ----------
+    field: xarray.DataArray
+        field to destagger
+    dim: str
+        dimension, one of {"x", "y", "generalVertical"}
+
+    Returns:
+    --------
+    dfield: xarray.DataArray
+        destaggered field with dimensions in {"x","y","generalVerticalLayer"}
+
+    """
     if dim == "x" or dim == "y":
         field_ = field
         field_[{dim: slice(1, None)}] = (
             field[{dim: slice(0, -1)}] + field[{dim: slice(1, None)}]
         ) * 0.5
         return field_
-    elif dim == "generalVerticalLayer":
-        hhl_k0 = field[{dim: slice(0, -1)}].assign_coords(
-            {dim: field[{dim: slice(0, -1)}].generalVerticalLayer}
+    elif dim == "generalVertical":
```

Comment:
I would restrict it preliminarily to the use case that we have for the COSMO and ICON data, where I think that destaggering from "generalVertical" to "generalVerticalLayer" is sufficient. It can be generalized, if a use case occurs. Rather could we implement a "stagger" operator in addition (if we need that).